lib/rubocop/cop/style/explicit_block_argument.rb

```diff
@@ -71,7 +71,10 @@ def yielding_arguments?(block_args, yield_args)
         def add_block_argument(node, corrector)
           if node.arguments?
             last_arg = node.arguments.last
-            corrector.insert_after(last_arg, ', &block') unless last_arg.blockarg_type?
+            arg_range = range_with_surrounding_comma(last_arg.source_range, :right)
+            replacement = ' &block'
+            replacement = ",#{replacement}" unless arg_range.source.end_with?(',')
+            corrector.insert_after(arg_range, replacement) unless last_arg.blockarg_type?
```

Comment:
I realize you didn't write that, but is `unless last_arg.blockarg_type?` useful here?

Reply:
Yes, it's required for this test to pass:

https://github.com/rubocop-hq/rubocop/blob/4943d5005b44c61973910b77adbb5fa42209bbfd/spec/rubocop/cop/style/explicit_block_argument_spec.rb#L36-L49

Here's the failure without it:

```
$ be rspec spec/rubocop/cop/style/explicit_block_argument_spec.rb:36
Run options: include {:focus=>true, :locations=>{"./spec/rubocop/cop/style/explicit_block_argument_spec.rb"=>[36]}}

Randomized with seed 48791
F

Failures:

  1) RuboCop::Cop::Style::ExplicitBlockArgument correctly corrects when method already has an explicit block argument
     Failure/Error: expect(new_source).to eq(correction)

       expected: "def m(&block)\n  items.something(&block)\nend\n"
            got: "def m(&block, &block)\n  items.something(&block)\nend\n"

       (compared using ==)

       Diff:
       @@ -1,4 +1,4 @@
       -def m(&block)
       +def m(&block, &block)
          items.something(&block)
        end

     # ./lib/rubocop/rspec/expect_offense.rb:164:in `expect_correction'
     # ./spec/rubocop/cop/style/explicit_block_argument_spec.rb:44:in `block (2 levels) in <top (required)>'

Finished in 0.05871 seconds (files took 1.67 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/rubocop/cop/style/explicit_block_argument_spec.rb:36 # RuboCop::Cop::Style::ExplicitBlockArgument correctly corrects when method already has an explicit block argument

Randomized with seed 48791
```

Reply:
Oh, I understand, thanks and sorry :-)